### DISCUSSION TITLE
typed tf Input type

### DISCUSSION THREAD

**Discussion by pgg** [vladmandic/human#269] [General]
Hi Vlad, how do I get/use the typed version of tf in human? the detect method errors as below:
```
import * as tfn from '@tensorflow/tfjs-node';
import { Human } from '@vladmandic/human';
   
const human = new Human();
const tf: typeof tfn = human.tf;
const tensor = tf.tidy(() => tf.node.decodeImage(image).toFloat().expandDims(0));
await human.detect(tensor);
```
Argument of type 'Tensor<Rank>' is not assignable to parameter of type 'Input'.
  Type 'import("@tensorflow/tfjs-core/dist/tensor").Tensor<import("@tensorflow/tfjs-core/dist/types").Rank>' is not assignable to type 'import("{my project path}/node_modules/@vladmandic/human/types/human").Tensor<import("{my project path}/node_modules/@vladmandic/human/types/human").Rank>'.
    Property 'isDisposedInternal' is protected but type 'Tensor<R>' is not a class derived from 'Tensor<R>'.ts(2345)

**Comment by vladmandic** [comment]
no easy solution at the moment i'm affraid.  
currently i'm bunding `tfjs` **typedefs** in a very hacky way as `tfjs` is not (yet, work is ongoing) `tsc` 4.x compatible, so i cannot generate them cleanly the way i'd like to.  
once `tfjs` update is complete, i'll switch to new way of generating **typedefs** and then this should work.

**Reply by pgg** [reply]
ok thanks
